### ISSUE TITLE
🐛 Submit does not navigate to error section every time

### ISSUE THREAD

**Issue by mnigh** [GCTC-NTGC/gc-digital-talent#8846]
## 🐛 Bug

When there are errors in the form and the form is submitted, the window should navigate to the error section every time the user submits.

## 🦋 Expected Behaviour

If there are errors in the form, the window should navigate to the error section every time.

## 📋 Steps to Reproduce

1. Navigate to career timeline
2. Add a new work experience
3. Fill out all required fields except **Additional details**
4. Enter **Save and return to my career timeline**
5. Observe the scroll
6. Fill out all fields except My role
7. Enter **Save and return to my career timeline**
8. Observe failure to scroll to error section

## 🎥 Screen recording

https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/fa56cb91-2681-45b5-b5ad-34b5d53c1d92

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the bug was addressed.

- [ ] when there are errors in the form, the window navigates to the error section every time

**Comment by petertgiles** [comment]
It looks like, in the video, it scrolled properly the first time.  It was when you changed the error condition to a different field it didn't scroll properly then.  Is that the problem - it doesn't scroll properly the second time you try to submit?

**Comment by mnigh** [comment]
> It looks like, in the video, it scrolled properly the first time. It was when you changed the error condition to a different field it didn't scroll properly then. Is that the problem - it doesn't scroll properly the second time you try to submit?

Yes.
